prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,6 +204,56 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Get a sortable list of facilities for multiple deliveries
+     *
+     * @param request HTML request body.
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilitiesBulk")
+    fun getDeliveryFacilitiesBulk(
+        @HttpTrigger(
+            name = "getDeliveryFacilitiesBulk",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/facilities"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        try {
+            return run {
```

Comment:
- How is authentication happening?
- How is it being ensured that the user has the rights to "see" the requested report_ids?

Is this a giant yawning security hole?

Reply:
Added a beefy security check now